webapp/graphite/render/functions.py

```diff
@@ -2631,12 +2631,14 @@ def threshold(requestContext, value, label=None, color=None):
 
   return [series]
 
-def transformNull(requestContext, seriesList, default=0):
+def transformNull(requestContext, seriesList, default=0, keySeries=None):
   """
-  Takes a metric or wild card seriesList and an optional value
-  to transform Nulls to. Default is 0. This method compliments
-  drawNullAsZero flag in graphical mode but also works in text only
-  mode.
+  Takes a metric or wild card seriesList and an optional value to transform
+  Nulls to. Default is 0. The optional keySeries, if specified, is a metric
+  or wild card series list that governs which data intervals nulls should be
+  transformed. If specified, nulls are transformed only in intervals where
+  the same interval in any of keySeries is non-null. This method compliments
+  drawNullAsZero flag in graphical mode but also works in text only mode.
```

Comment:
`drawNullAsZero flag` to `the drawNullAsZero function` and `text only` to `text-only`